src/main/java/com/taskadapter/redmineapi/AttachmentManager.java

```diff
@@ -7,198 +14,144 @@
-                "uploadStream");
-        final String token;
-        try {
-            token = transport.upload(wrapper);
-            final Attachment result = AttachmentFactory.create();
-            result.setToken(token);
-            result.setContentType(contentType);
-            result.setFileName(fileName);
-            return result;
-        } catch (RedmineException e) {
-            unwrapException(e, "uploadStream");
-            throw e;
-        }
-    }
+	/**
+	 * Delivers an {@link com.taskadapter.redmineapi.bean.Attachment} by its ID.
+	 * 
+	 * @param attachmentID
+	 *            the ID
+	 * @return the {@link com.taskadapter.redmineapi.bean.Attachment}
+	 * @throws RedmineAuthenticationException
+	 *             thrown in case something went wrong while trying to login
+	 * @throws RedmineException
+	 *             thrown in case something went wrong in Redmine
+	 * @throws NotFoundException
+	 *             thrown in case an object can not be found
+	 */
+	public Attachment getAttachmentById(int attachmentID) throws RedmineException {
+		return transport.getObject(Attachment.class, attachmentID);
+	}
 
-    /**
-     * Delivers an {@link com.taskadapter.redmineapi.bean.Attachment} by its ID.
-     *
-     * @param attachmentID the ID
-     * @return the {@link com.taskadapter.redmineapi.bean.Attachment}
-     * @throws RedmineAuthenticationException thrown in case something went wrong while trying to login
-     * @throws RedmineException        thrown in case something went wrong in Redmine
-     * @throws NotFoundException       thrown in case an object can not be found
-     */
-    public Attachment getAttachmentById(int attachmentID) throws RedmineException {
-        return transport.getObject(Attachment.class, attachmentID);
-    }
+	public void downloadAttachmentContent(Attachment issueAttachment, OutputStream stream) throws RedmineException {
+		transport.download(issueAttachment.getContentURL(), new CopyBytesHandler(stream));
+	}
 
-    public void downloadAttachmentContent(Attachment issueAttachment,
-                                          OutputStream stream) throws RedmineException {
-        transport.download(issueAttachment.getContentURL(),
-                new CopyBytesHandler(stream));
-    }
+	/**
+	 * Downloads the content of an {@link com.taskadapter.redmineapi.bean.Attachment} from the Redmine server.
+	 * 
+	 * @param issueAttachment
+	 *            the {@link com.taskadapter.redmineapi.bean.Attachment}
+	 * @return the content of the attachment as a byte[] array
+	 * @throws RedmineCommunicationException
+	 *             thrown in case the download fails
+	 */
+	public InputStream downloadAttachmentContent(Attachment issueAttachment) throws RedmineException {
+		final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+		downloadAttachmentContent(issueAttachment, baos);
+		try {
+			baos.close();
+		} catch (IOException e) {
+			throw new RedmineInternalError();
```

Comment:
You are loosing the cause of the exception. What should a library user do with this exception? He even do not see any relevant details about it.